src/pineko/evolve.py

```diff
@@ -173,28 +187,26 @@ def evolve_grid(
     )
     check.check_grid_and_eko_compatible(grid, operators, xif, max_as, max_al)
     # rotate to evolution (if doable and necessary)
-    if np.allclose(operators.rotations.inputpids, br.flavor_basis_pids):
+    if np.allclose(operators.bases.inputpids, br.flavor_basis_pids):
         eko.io.manipulate.to_evol(operators)
     # Here we are checking if the EKO contains the rotation matrix (flavor to evol)
-    elif not np.allclose(operators.rotations.inputpids, br.rotate_flavor_to_evolution):
+    elif not np.allclose(operators.bases.inputpids, br.rotate_flavor_to_evolution):
         raise ValueError("The EKO is neither in flavor nor in evolution basis.")
-    # This is really the facto scale grid only for scheme A and C
-    muf2_grid = operators.mu2grid
     # PineAPPL wants alpha_s = 4*pi*a_s
     # remember that we already accounted for xif in the opcard generation
     evmod = eko.couplings.couplings_mod_ev(opcard.configs.evolution_method)
     # Couplings ask for the square of the masses
-    quark_masses = [(x.value) ** 2 for x in tcard.quark_masses]
+    thresholds_ratios = np.power(tcard.heavy.matching_ratios, 2.0)
+    for q in range(tcard.couplings.max_num_flavs + 1, 6 + 1):
+        thresholds_ratios[q - 4] = np.inf
     sc = eko.couplings.Couplings(
         tcard.couplings,
         tcard.order,
         evmod,
-        quark_masses,
-        hqm_scheme=tcard.quark_masses_scheme,
-        thresholds_ratios=np.power(list(iter(tcard.matching)), 2.0),
+        masses=[(x.value) ** 2 for x in tcard.heavy.masses],
+        hqm_scheme=tcard.heavy.masses_scheme,
+        thresholds_ratios=thresholds_ratios,
```

Comment:
```suggestion
        thresholds_ratios=thresholds_ratios.tolist(),
```
I you want for future ourself, just to be sure.